CLI/add_solver.py

```diff
@@ -100,6 +100,11 @@ def parser_function() -> argparse.ArgumentParser:
         sys.exit(-1)
     shutil.copytree(solver_source, solver_directory, dirs_exist_ok=True)
 
+    if not sash.check_solver_executable_permission(solver_directory):
+        print("ERROR: Missing execution permission for binary of solver "
```

Comment:
This can be a warning instead of an error. Also good to report to the user which file is missing the executable permission.